stacks/ingress-nginx/README.md

```diff
@@ -0,0 +1,167 @@
+
+Typical ingress resource for `Nginx` looks like below:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: ingress-echo
+  namespace: backend
+spec:
+  ingressClassName: nginx
+  rules:
+    - host: echo.starter-kit.online
+      http:
+        paths:
+          - path: /
+            pathType: Prefix
+            backend:
+              service:
+                name: echo
+                port:
+                  number: 8080
+```
+
+Explanations for the above configuration:
+
+- `spec.rules`: A list of host rules used to configure the Ingress. If unspecified, or no rule matches, all traffic is sent to the default backend.
+- `spec.rules.host`: Host is the fully qualified domain name of a network host (e.g.: `echo.starter-kit.online`).
+- `spec.rules.http`: List of http selectors pointing to backends.
+- `spec.rules.http.paths`: A collection of paths that map requests to backends. In the above example the `/` path prefix is matched with the `echo` backend `service`, running on port `8080`.
+
+The above ingress resource tells `Nginx` to `route` each `HTTP request` that is using the `/` prefix for the `echo.starter-kit.online` host, to the `echo` backend `service` running on port `8080`. In other words, every time you make a call to `http://echo.starter-kit.online/` the request and reply will be served by the `echo` backend `service` running on port `8080`.
+
+### Upgrading the Nginx Ingress Chart
+
+```console
+helm upgrade [RELEASE_NAME] [CHART]
+```
+
+See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documentation.
+
+### Upgrading With Zero Downtime in Production
+
+By default the ingress-nginx controller has service interruptions whenever it's pods are restarted or redeployed. In order to fix that, see the excellent blog post by Lindsay Landry from Codecademy: [Kubernetes: Nginx and Zero Downtime in Production](https://medium.com/codecademy-engineering/kubernetes-nginx-and-zero-downtime-in-production-2c910c6a5ed8).
+
+### Migrating from stable/nginx-ingress
+
+There are two main ways to migrate a release from `stable/nginx-ingress` to `ingress-nginx/ingress-nginx` chart:
+
+1. For Nginx Ingress controllers used for non-critical services, the easiest method is to uninstall the old release and install the new one.
+2. For critical services in production that require zero-downtime, you will want to:
+
+- Install a second Ingress controller.
+- Redirect your DNS traffic from the old controller to the new controller.
+- Log traffic from both controllers during this changeover.
+- Uninstall the old controller once traffic has fully drained from it.
+- For details on all of these steps see [Upgrading With Zero Downtime in Production](#upgrading-with-zero-downtime-in-production).
+
+### Uninstalling the Nginx Ingress Controller
+
+To delete your installation of NGINX Ingress Controller, please run the following `Helm` command:
+
+```console
+helm uninstall ingress-nginx -n ingress-nginx
```

Comment:
should we also add `kubectl delete ns ingress-nginx` as part of the instructions?

Reply:
Good point. Should be fixed in the latest commit (added a note in the README).